### DISCUSSION TITLE
Attemp to build a RAD almost succeeded ....

### DISCUSSION THREAD

**Discussion by crystalct** [frntc/RAD#6] [General]
... but it doesn't work correctly.
Used with a Rapsberry A+.
The menu starts but the graphic look degrades over time.
Starting mode selected inside rad.cfg is ignored, it starts menu always.
Every PRG selected: back to c64 and immediately back to menu.
Exit to c64 using X... back to menu immediatetely.
Components used (by mouser):
595-SN74CBTD3861DWR
771-74HCT30D-Q100J
771-74LVC245AD-T
771-LVC257AD118
771-74LVC573AD-T

Used a C64 C and C64 breadboard with 250425 assy (original PLA) and external USB VCC supply.

**Comment by frntc** [comment]
Can you check if you have a short circuit at your menu button? Would be an explanation...

**Reply by crystalct** [reply]
it's the first thing i did... sadly no.
with a tester I tried all the welding of the components, following the diagram on a gerber viewer... all ok.

**Reply by crystalct** [reply]
just tested again.... i measured 17K ohm.

**Reply by frntc** [reply]
so pin 11 of the outer LVC257 is 5V when the button is not pressed? between this pin and pin1 of the RPi-connector you should measure 10kR (and no connection to GND when you don't press the button)

**Reply by crystalct** [reply]
To do that test, must cartridge be connected to c64 or just to +5V from USB?

**Reply by frntc** [reply]
You can measure the resistance without power/C64. The +5V is a bit cumbersome to measure, you can also measure it at pin 2 of the menu button that's a bit easier:
![pin2button](https://user-images.githubusercontent.com/48584231/226328619-aefadbe9-53bd-4365-bb2a-1c16d59f37e2.jpg)

**Comment by crystalct** [comment]
Pin11 is 0.45V when the button is not pressed. No connection to GND when it's not pressed. 10K ohm between pin11 and pin 1 of the RPi-connector.

![bitmoji](https://sdk.bitmoji.com/render/panel/3b753aef-6efb-4ea1-8460-7838285d671f-888af1ad-9072-4a68-8915-4e015bf57555-v1.png?transparent=1&palette=1&width=246)

**Reply by frntc** [reply]
ok, something's pulling the signal down. does the voltage change (go to 0V) if you press the button?

I'd check (in the following order):
- look for solder bridges at the 10k resistors and the LVC257
- desolder the button and see if it behaves the same
- change the LVC257

**Comment by crystalct** [comment]
BINGO!!!
R2 wasn't well soldered!!!

Thanks a lot!

**Comment by frntc** [comment]
Great, have fun with the RAD!

**Comment by SensoryTrash** [comment]
I am trying to get my RAD working and wondered what source you used for the chips...

74HCT30	- 74HCT30D-Q100J Digikey
74CBTD3861 - SN74CBTD3861DWR Ebay
74LVC245 - 74LVC245AD,118	Digikey
74LVC257 - 74LVC257ADB,112 Digikey  im thinking this one is the problem.
74LVC573 - 74LVC573AD,118 Digikey

**Reply by frntc** [reply]
The LVC/HCT you get basically everywhere, the CBTD were a bit more difficult to find but, for example, Digikey and Mouser have them on stock. If sourcing the LVC257 is a problem, you can also use a LVC157. (I never bought components via ebay, so can't recommend anything here)

**Comment by futureretrofusion** [comment]
I found that the RAD had issues with the 573AD but it works properly with
573A please see pics attached.

**Comment by crystalct** [comment]
What Pics?

**Comment by futureretrofusion** [comment]
https://drive.google.com/file/d/1X329gw_v8hHRWpqZ4ss8fF0YP1WzGafr/view?usp=sharing

https://drive.google.com/file/d/1-V6hMABnD79BqmY0GjDdk3KB6Nmg0HNe/view?usp=sharing

**Comment by SensoryTrash** [comment]
Thanks to everyone for the information... does anyone have a good source for this chip as i am fearful of ordering the wrong one... I am located in Canada...

**Reply by frntc** [reply]
The letters after 573A only encode housing and packaging, there should be no difference. I also had LVCs from different manufacturers and all worked. Maybe that particular one is not working properly?

**Comment by SensoryTrash** [comment]
Iok i got it i had fitted the components that are not required and just removed them and it work fantastically thank you so much to everyone for being patient with a noob like me
